iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -103,7 +106,7 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     explicit optional(in_place_t, Targs&&... args) noexcept;
 
     /// @brief The destructor will call the destructor of T if a value is set.
-    ~optional() noexcept;
+    virtual ~optional() noexcept;
```

Comment:
Please remove this! The optional is used in the shared memory and this would introduce a bug and weird segfaults.

The problem is that the `virtual` keyword introduces a vtable which is in the end a list of function pointers. But the function pointers are only valid in one process space. Another process may has the same function stored in a different place and then we either have undefined behavior or a crash.

Reply:
@elfenpiff My understanding is that a `vtable` is a class level entity, and the entries shall be same across the usages.
we can remove `virtual` only if the class is made a final. otherwise, we need to suppress `AutosarC++19_03-A12-4-2'

Reply:
@saif-at-github Making the class `final` should not pose a problem for most `cxx` constructs. Can you please try it out.

Reply:
@elfenpiff that works for sure. thanks.